csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/CXX_Api_Sample.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+//
+
+#include <assert.h>
+#include <vector>
+#include <onnxruntime_cxx_api.h>
+
+int main(int argc, char* argv[]) {
+  //*************************************************************************
+  // initialize  enviroment...one enviroment per process
+  // enviroment maintains thread pools and other state info
+  Ort::Env env(ORT_LOGGING_LEVEL_WARNING, "test");
+
+  // initialize session options if needed
+  Ort::SessionOptions session_options;
+  session_options.SetThreadPoolSize(1);
+
+  // If onnxruntime.dll is built with CUDA enabled, we can uncomment out this line to use CUDA for this
+  // session (we also need to include cuda_provider_factory.h above which defines it)
+  // #include "cuda_provider_factory.h"
+  // OrtSessionOptionsAppendExecutionProvider_CUDA(session_opsions, 1);
+
+  // Sets graph optimization level
+  // Available levels are 
+  // 0 -> To disable all optimizations
+  // 1 -> To enable basic optimizations (Such as redundant node removals)
+  // 2 -> To enable all optimizations (Includes level 1 + more complex optimizations like node fusions)
+  session_options.SetGraphOptimizationLevel(1);
+
+  //*************************************************************************
+  // create session and load model into memory
+  // using squeezenet version 1.3
+  // URL = https://github.com/onnx/models/tree/master/squeezenet
+  const wchar_t* model_path = L"squeezenet.onnx";
+  Ort::Session session(env, model_path, session_options);
+
+  //*************************************************************************
+  // print model input layer (node names, types, shape etc.)
+  Ort::Allocator allocator=Ort::Allocator::Create_Default();
+
+  // print number of model input nodes
+  size_t num_input_nodes=session.GetInputCount();
+  std::vector<const char*> input_node_names(num_input_nodes);
+  std::vector<int64_t> input_node_dims;  // simplify... this model has only 1 input node {1, 3, 224, 224}.
+                                         // Otherwise need vector<vector<>>
+
+  printf("Number of inputs = %zu\n", num_input_nodes);
+
+  // iterate over all input nodes
+  for (int i = 0; i < num_input_nodes; i++) {
+    // print input node names
+    char* input_name=session.GetInputName(i, allocator);
+    printf("Input %d : name=%s\n", i, input_name);
+    input_node_names[i] = input_name;
+
+    // print input node types
+    Ort::TypeInfo type_info=session.GetInputTypeInfo(i);
+    auto tensor_info=type_info.GetTensorTypeAndShapeInfo();
+
+    tensor_info.NonConstTest();
```

Comment:
A comment around NonConstTest() would be useful (i.e. what it does and is it an optional test).

Reply:
Oops, that was meant to be deleted, good catch!